### PR TITLE
refactor(Authoring): Create branch path authoring component

### PR DESCRIPTION
**Pull request by geoffreykwan**
## Changes
- Created BranchPathAuthoringComponent
- Create Branch view now uses the BranchPathAuthoringComponent

## Test
- Make sure the create branch form still works properly and validates properly
- Test out all the criteria types
  - Workgroup ID
  - Score
  - Choice Chosen
  - Random
  - Tag

### CODE REVIEW

File: src/assets/wise5/authoringTool/add-branch/add-branch.component.ts [hunk 1]
```diff
@@ -13,30 +13,34 @@ import { MatInputModule } from '@angular/material/input';
...
   imports: [
+    BranchPathAuthoringComponent,
     CommonModule,
     FlexLayoutModule,
     MatButtonModule,
     MatFormFieldModule,
     MatInputModule,
     MatSelectModule,
     ReactiveFormsModule,
+    RouterModule,
```
Comment (hirokiterashima) [inline src/assets/wise5/authoringTool/add-branch/add-branch.component.ts:41]: Where do we use this import?

File: src/assets/wise5/services/branchService.ts [hunk 2]
```diff
@@ -5,6 +5,9 @@ import { PathService } from './pathService';
 
 @Injectable()
 export class BranchService {
+  readonly CHOICE_CHOSEN: string = 'choiceChosen';
+  readonly SCORE: string = 'score';
```
Comment (hirokiterashima) [inline src/assets/wise5/services/branchService.ts]: Should these variables be in a separate file, outside of BranchService? Having to depend on BranchService just to use these variables like we do now in BranchPathAuthoringComponent doesn't seem ideal.

**Review by hirokiterashima — COMMENTED** [review]
Works as described. I added some comments inline about code. Lmk what you think.

File: src/assets/wise5/authoringTool/add-branch/add-branch.component.ts (hunk 1 above)
Comment (geoffreykwan) [reply src/assets/wise5/authoringTool/add-branch/add-branch.component.ts:41]: The RouterModule is required for the cancel button routerlink in the template.

File: src/assets/wise5/services/branchService.ts (hunk 2 above)
Comment (geoffreykwan) [reply src/assets/wise5/services/branchService.ts]: Good idea, I've made the changes.

**Review by hirokiterashima — APPROVED** [review]
LGTM. Looks like you can remove branchService dependency from BranchPathAuthoringComponent?

**Pull request merged**